src/command.c

```diff
@@ -212,8 +212,10 @@ static bool cmd_jtag_scan(target_s *target, int argc, const char **argv)
 	(void)argc;
 	(void)argv;
 
-	if (platform_target_voltage())
-		gdb_outf("Target voltage: %s\n", platform_target_voltage());
+	/* Call the sampling function only once, check for NULL, check for unimplemented and suppress the output */
+	const char *const target_voltage = platform_target_voltage();
+	if (target_voltage && (!strcmp(target_voltage, "Unknown")))
+		gdb_outf("Target voltage: %s\n", target_voltage);
```

Comment:
Given we're defining `platform_target_voltage()` to be non-null, think we can get away with simplifying this to just:
```suggestion
	gdb_outf("Target voltage: %s\n", platform_target_voltage());
```

And likewise for the other command.c invocations - we think displaying "Target voltage: Unknown" on platforms that don't implement this is very acceptable.

Reply:
> I can't drop the runtime NULL check in good faith

None of the in-tree platforms are returning NULL now, and this adds a comment to the header explaining valid values. It's not on you at that point, dropping the NULL check is completely safe. It is on out-of-tree platforms to pay attention when they update and to use release notes. If this wasn't true, we'd be permanently shouldered with all the bad decisions of out-of-tree platforms forever and would never be able to change and evolve the project.

> Displaying an extra string changes UI for users of GDB CLI, this is outside of my memory UB bugfix scope PR (I'd like a second opinion about that).

It is in-scope because it makes everything much more consistent across the platforms and cleans up after the NULL pointer fix, thus is in scope. From the project's perspective based on the original intention of this from the native platform, it's actually an error that the target voltage display is not on all platforms, making it a bugfix based on the intention that this string should be displayed.